internal/operator/rolebinding/rolebinding.go

```diff
@@ -0,0 +1,137 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package rolebinding
+
+import (
+	"fmt"
+	"strings"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/common"
+	druiderr "github.com/gardener/etcd-druid/internal/errors"
+	"github.com/gardener/etcd-druid/internal/operator/component"
+	"github.com/gardener/etcd-druid/internal/utils"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ErrGetRoleBinding indicates an error in getting the role binding resource.
+	ErrGetRoleBinding druidv1alpha1.ErrorCode = "ERR_GET_ROLE_BINDING"
+	// ErrSyncRoleBinding indicates an error in syncing the role binding resource.
+	ErrSyncRoleBinding druidv1alpha1.ErrorCode = "ERR_SYNC_ROLE_BINDING"
+	// ErrDeleteRoleBinding indicates an error in deleting the role binding resource.
+	ErrDeleteRoleBinding druidv1alpha1.ErrorCode = "ERR_DELETE_ROLE_BINDING"
+)
+
+type _resource struct {
+	client client.Client
+}
+
+// New returns a new role binding operator.
+func New(client client.Client) component.Operator {
+	return &_resource{
+		client: client,
+	}
+}
+
+// GetExistingResourceNames returns the name of the existing role binding for the given Etcd.
+func (r _resource) GetExistingResourceNames(ctx component.OperatorContext, etcd *druidv1alpha1.Etcd) ([]string, error) {
+	resourceNames := make([]string, 0, 1)
+	objectKey := getObjectKey(etcd)
+	rb := &rbacv1.RoleBinding{}
+	if err := r.client.Get(ctx, objectKey, rb); err != nil {
```

Comment:
This could be metadata-only get instead of the full object.

Reply:
done